src/app/collective-rewards/utils/chartUtils.ts

```diff
@@ -9,3 +10,154 @@ export const formatShort = (n: number) => {
   return `${n.toLocaleString()}`
 }
+
+/**
+ * Generic function to fill gaps between dates with the last known value
+ * Creates daily data points from sparse data using functional approach
+ */
+const interpolateDataByDay = <T extends { day?: number; cycleStart?: string }>(
+  data: T[],
+  getTimestamp: (item: T) => number,
+  createInterpolatedItem: (day: number, lastKnownItem: T) => T,
+): T[] => {
+  if (!data || data.length === 0) return []
+
+  const sortedData = [...data].sort((a, b) => getTimestamp(a) - getTimestamp(b))
+  const startDate = getTimestamp(sortedData[0])
+  const endDate = getTimestamp(sortedData[sortedData.length - 1])
+
+  const dataMap = new Map(sortedData.map(item => [getTimestamp(item), item]))
+
+  const totalDays = Math.floor((endDate - startDate) / ONE_DAY_IN_SECONDS) + 1
+  const allDays = Array.from({ length: totalDays }, (_, i) => startDate + i * ONE_DAY_IN_SECONDS)
+
+  let lastKnownItem = sortedData[0]
+
+  return allDays.map(day => {
+    const actualData = dataMap.get(day)
+
+    if (actualData) {
+      lastKnownItem = actualData
+      return actualData
+    }
+
+    return createInterpolatedItem(day, lastKnownItem)
+  })
+}
+
+/**
+ * Fills gaps between dates with the last known value for DailyAllocationItem
+ */
+const interpolateDailyBackingData = (data: DailyAllocationItem[]): DailyAllocationItem[] => {
+  return interpolateDataByDay(
+    data,
+    item => item.day,
+    (day, lastKnownItem) => ({
+      id: `interpolated-${day}`,
+      day,
+      totalAllocation: lastKnownItem.totalAllocation,
+    }),
+  )
+}
+
+/**
+ * Fills gaps between dates with the last known value for CycleRewardsItem
+ */
+const interpolateCycleRewardsData = (data: CycleRewardsItem[]): CycleRewardsItem[] => {
+  return interpolateDataByDay(
+    data,
+    item => Number(item.cycleStart),
+    (day, lastKnownItem) => ({
+      ...lastKnownItem,
+      id: `interpolated-${day}`,
+      cycleStart: day.toString(),
+    }),
+  )
+}
+
+/**
+ * Calculate USD value from RIF and RBTC rewards
+ */
+const calculateRewardsUSD = (
+  rifAmountWei: string,
+  rbtcAmountWei: string,
+  rifPrice: number,
+  rbtcPrice: number,
+): number => {
+  const rifTokens = Number(rifAmountWei) / WEI_DIVISOR
+  const rbtcTokens = Number(rbtcAmountWei) / WEI_DIVISOR
+  return rifTokens * rifPrice + rbtcTokens * rbtcPrice
+}
+
+/**
+ * Transform backing data to BackingPoint format with interpolation
+ */
+const transformBackingData = (backingData: DailyAllocationItem[]): BackingPoint[] => {
+  if (!backingData || !Array.isArray(backingData)) return []
+
+  const interpolatedData = interpolateDailyBackingData(backingData)
+  return interpolatedData.map(item => ({
+    day: new Date(item.day * 1000),
+    backing: BigInt(Math.floor(Number(item.totalAllocation) / WEI_DIVISOR)),
+    backingWei: BigInt(item.totalAllocation),
+  }))
+}
+
+/**
+ * Transform cycle rewards data to RewardsPoint format with interpolation
+ */
+const transformRewardsData = (
```

Comment:
when transforming rewardData, we should sum up rewards for each cycle, so for instance, if at cycle 1 we distributed 10 RIF and at cycle 2 we distributed 10 RIF, the rewards data are cumulative, meaning that at cycle 1, we'll show 10 RIF while at cycle 2, we'll show 20 RIF (cycle 1 rewards + cycle 2 rewards)